component/component/provider.jsonnet

```diff
@@ -248,7 +255,6 @@ local controllerConfigRef(config) =
           }
         ),
       },
-      sa,
```

Comment:
Why is the SA object removed here?
Furthermore the SA that is not applied anymore is still referenced in the `rolebinding` object

Reply:
Because Crossplane 1.14 will automatically deploy the SA. It's still referenced in the `rolebinding` because we want our roles to be applied to it.